.husky/post-checkout

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+git config commit.template .git/templatemessage
+TICKETID=`git reflog | awk 'NR==1{ print $8; exit }' | LC_ALL=en_US.utf8 grep -oP '\d{3,5}'`
+echo "[POST_CHECKOUT] Setting template commit to #$TICKETID"
+echo "#$TICKETID " > ".git/templatemessage"
```

Comment:
could you explain what that does? :)

Reply:
when you checkeout a branch it takes the branch name and tries to get the ticket id. this then defaults the start of the commit message to #NNNN to help improve compliance in commit messages

Reply:
Fallback?
Is this executed Locally?

Reply:
its executed on any machine, local or not. In a workflow for example it will also execute, unless overriden (which you can also do). What would you look for in a fallback?

Reply:
when you say "defaults" the commit message - does that impact what people see in their git clients, like sourcetree or gitfork? or where is this "defaulted"?

if it doesn't find an id (by the way, what is it looking for exactly? I cant figure out that grep logic...) it just prints `"# "` into .git/templatemessage?

Reply:
never mind - im seeing this happens upon stage in gitfork, at the latest upon clicking into the commit message field - unless the field already has content in it

Reply:
tested this some more: if it does not find an id it seems to fail on line 5 and subsequently never update .git/templatemessage again. to test that, I first checked out your branch (175) and played around with the auto-completion. Then, I created a branch off of `develop`, switched to and merged your code in, afterwards I made a change and checked again. It would still autocomplete "#175 ".

same happened when i renamed your 175 branch to something without an id which i tried before the branch off of develop.

- [ ] add fallback for branches that don't have ids in them (e.g. created by external developers in their forked repos)